pkg/cluster/storageaccounts.go

```diff
@@ -5,18 +5,91 @@ package cluster
 
 import (
 	"context"
+	"strings"
 
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 )
 
+// enableServiceEndpoints should enable service endpoints on
+// subnets for storage account access
+func (m *manager) enableServiceEndpoints(ctx context.Context) error {
+	subnets := []string{
+		m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID,
+	}
+
+	for _, wp := range m.doc.OpenShiftCluster.Properties.WorkerProfiles {
+		subnets = append(subnets, wp.SubnetID)
+	}
+
+	for _, subnetId := range subnets {
+		subnet, err := m.subnet.Get(ctx, subnetId)
+		if err != nil {
+			return err
+		}
+
+		var changed bool
+		for _, endpoint := range api.SubnetsEndpoints {
+			var found bool
+			if subnet != nil && subnet.ServiceEndpoints != nil {
+				for _, se := range *subnet.ServiceEndpoints {
+					if strings.EqualFold(*se.Service, endpoint) &&
+						se.ProvisioningState == mgmtnetwork.Succeeded {
+						found = true
+					}
+				}
+			}
+			if !found {
+				if subnet.ServiceEndpoints == nil {
+					subnet.ServiceEndpoints = &[]mgmtnetwork.ServiceEndpointPropertiesFormat{}
+				}
+				*subnet.ServiceEndpoints = append(*subnet.ServiceEndpoints, mgmtnetwork.ServiceEndpointPropertiesFormat{
+					Service:   to.StringPtr(endpoint),
+					Locations: &[]string{"*"},
+				})
+				changed = true
+			}
+		}
+		if changed {
+			err := m.subnet.CreateOrUpdate(ctx, subnetId, subnet)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// migrateStorageAccounts should re-deploy storage account with encryption,
```

Comment:
This comment seems inaccurate since we don't enable encryption for storage accounts without it: https://github.com/Azure/ARO-RP/pull/1865/files#diff-a4c886e1a4a25fceb3a8ca3064fc76f692b6f1cbbc191bd31a6778a3aabf0fe5R134

Reply:
Updated.